dialogue-test-common/src/main/java/com/palantir/dialogue/TestResponse.java

```diff
@@ -72,20 +72,20 @@ public TestResponse code(int value) {
     public void close() {
         checkNotClosed();
         try {
-            closeCalled = Optional.of(new SafeRuntimeException("Close called here"));
+            closeCalled.compareAndSet(false, true);
```

Comment:
boolean cas isn't necessary, once this is called we're always in the `true` state ;-)